### PR TITLE
Added Seq.skipLast(long n) and Seq.limitLast(long maxSize)

### PR DESCRIPTION
**Pull request by tlinkowski**
See #313.

### CODE REVIEW

File: src/main/java/org/jooq/lambda/SequentialSpliterator.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * Special subinterface of <code>Spliterator</code> that provides default implementations
+ * for all the methods that are not relevant to sequential traversal.
+ *
+ * @see java.util.stream.BaseStream#sequential
+ *
+ * @author Tomasz Linkowski
+ */
+@FunctionalInterface
+public interface SequentialSpliterator<T> extends Spliterator<T> {
```
Comment (lukaseder) [inline src/main/java/org/jooq/lambda/SequentialSpliterator.java]: Hmm, interesting - good idea. Perhaps we should put that in a separate issue, though. I'll decide later, when it's clear if this type really needs to be public

Comment (tlinkowski) [inline src/main/java/org/jooq/lambda/SequentialSpliterator.java]: For now, I just placed it inside `TailBuffer`. But maybe it doesn't need to be public, I must say I just added it as public without much consideration.

Comment (lukaseder) [inline src/main/java/org/jooq/lambda/SequentialSpliterator.java]: I see. As always, I prefer this not to be public, then :)

**Comment by lukaseder** [comment]
The PR got closed by github due to a recent rename of the main branch.

**Comment by lukaseder** [comment]
I'm not sure we need this in the API

**Pull request closed without merging**